test/e2e/storage/testsuites/subpath.go

```diff
@@ -110,7 +110,7 @@ func (s *subPathTestSuite) execTest(driver drivers.TestDriver, pattern testpatte
 			needsCleanup = true
 
 			// Setup test resource for driver and testpattern
-			resource := subPathTestResource{}
+			resource = subPathTestResource{}
```

Comment:
this small change is actually incredibly crucial. I don't know why go/ginkgo behaves like this but if you do `:=` you get a NEW locally scoped variable `resource` and doesn't complain even though you have a larger scoped variable `resource` already defined. You have to use `=` to explicitly set the variable in the larger scope so that the `AfterEach` block can use it.

Reply:
it's golang shadowing :(

Reply:
@davidz627 

I've just noticed it and would like to point out. Thank you for the fix.

Reply:
Does the resource evaluation in the `AfterEach` pass by value or reference?

Reply:
the method `cleanupResource` takes a pointer receiver so it will pass the pointer to `resource`.